src/app/components/Map/Map.tsx

```diff
@@ -159,10 +181,11 @@ export default function MapBox(): JSX.Element {
         },
       })
     }
-
-    if (map.current?.getLayer('end')) {
-      map.current.getSource('end').setData(endData)
-    } else {
+    const mapEndSource = map.current?.getSource('end')
+    if (mapEndSource?.type === 'geojson') {
+      mapEndSource.setData(endData)
+    } else if (location2) {
+      const coordinates = location2
```

Comment:
Actually, I think you don't neccessarily need this, just do it like you did for the other source